dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/ContentStorageResource.java

```diff
@@ -0,0 +1,218 @@
+package com.dotcms.rest.api.v1.storage;
+
+import com.dotcms.repackage.com.google.common.annotations.VisibleForTesting;
+import com.dotcms.repackage.org.codehaus.jettison.json.JSONException;
+import com.dotcms.rest.InitDataObject;
+import com.dotcms.rest.ResponseEntityView;
+import com.dotcms.rest.WebResource;
+import com.dotcms.rest.annotation.NoCache;
+import com.dotcms.rest.api.MultiPartUtils;
+import com.dotcms.storage.ContentletMetadata;
+import com.dotcms.storage.ContentletMetadataAPI;
+import com.dotcms.storage.FileStorageAPI;
+import com.dotcms.storage.StoragePersistenceProvider;
+import com.dotcms.storage.StorageType;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.web.LanguageWebAPI;
+import com.dotmarketing.business.web.WebAPILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.PageMode;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageUtil;
+import io.vavr.Tuple2;
+import javax.validation.Valid;
+import org.apache.commons.io.FilenameUtils;
+import org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import org.glassfish.jersey.server.JSONP;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Resource to expose the storage and metadata
+ * @author jsanca
+ */
+@Path("/v1/storage")
+public class ContentStorageResource {
+
+    private final WebResource webResource;
+    private final FileStorageAPI fileStorageAPI;
+    private final MultiPartUtils multiPartUtils;
+    private final ContentStorageHelper helper;
+    private final ContentletMetadataAPI contentletMetadataAPI;
```

Comment:
Codacy found an issue: [Perhaps 'contentletMetadataAPI' could be replaced by a local variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)